Modules/MoonbeamMainModule.php

```diff
@@ -0,0 +1,87 @@
+<?php declare(strict_types = 1);
+
+/*  Idea (c) 2023 Nikita Zhavoronkov, nikzh@nikzh.com
+ *  Copyright (c) 2023 3xpl developers, 3@3xpl.com, see CONTRIBUTORS.md
+ *  Distributed under the MIT software license, see LICENSE.md  */
+
+/*  This is the main Moonbeam parachain module. It requires a moonbeam node to run. 
+ *  This module only handles EVM-like activity; moonbeam components, that can
+ *  only be accessed via Substrate API (such as staking & validator rewards) 
+ *  are processed by a separate module. */
+
+final class MoonbeamMainModule extends EVMMainModule implements Module
+{
+    function initialize()
+    {
+        // CoreModule
+        $this->blockchain = 'moonbeam';
+        $this->module = 'moonbeam-main';
+        $this->is_main = true;
+        $this->first_block_date = '2021-12-18';
+        $this->currency = 'glimmer';
+        $this->currency_details = ['name' => 'Glimmer', 'symbol' => 'GLMR', 'decimals' => 18, 'description' => null];
+
+        // EVMMainModule
+        $this->evm_implementation = EVMImplementation::geth;
+        $this->extra_features = [EVMSpecialFeatures::FeesToTreasury];
+        $this->special_addresses = ['the-void', 'treasury'];
+        $this->mempool_implemented = true;
+        $this->forking_implemented = false;
```

Comment:
Ah, forgot to mention that this should be set to `true` in all Moonbeam modules due to often reorgs happening in substrate chains, and moonbeam as well:

```
2023-10-18 20:10:48 [Relaychain] ✨ Imported #17780510 (0x48e0…00f9)    
2023-10-18 20:10:48 [Relaychain] ♻️  Reorg on #17780510,0x48e0…00f9 to #17780510,0x081a…9700, common ancestor #17780509,0xfdf3…22a8    
2023-10-18 20:10:48 [Relaychain] ✨ Imported #17780510 (0x081a…9700)    
2023-10-18 20:10:51 [🌗] 💤 Idle (21 peers), best: #4676959 (0x172f…aa4c), 
```